### ISSUE TITLE
compressed .wrl (.wrz) files silently fail to decompress

### ISSUE THREAD

**Issue by evils** [flathub/org.freecad.FreeCAD#11]
first encountered in the Nixpkgs package,
i had initially opened a [FreeCAD issue](https://github.com/FreeCAD/FreeCAD/issues/10803) because a friend of mine reproduced the problem on linux mint with flatpak
however there we failed to reproduce it on windows, ubuntu and in the appimage
therefore i filed a [Nixpkgs issue](https://github.com/NixOS/nixpkgs/issues/257313)
and am opening this issue

the issues linked above have more details and outline steps to reproduce the problem

a similar issue occurs in the Nix package for KiCad, but not in the flatpak package for KiCad

**Comment by hfiguiere** [comment]
> are we building coin wrong?

Yes and no. It's coin that is broken because whoever wrote the code to dlopen zlib doesn't grasp the concept of soversion. It's a classic seen far too many times.

It works when I tried to track what was wrong, because `--devel` use the Sdk as a runtime, SDK that has a `.so`.
